contracts/RemixChallenges.sol

```diff
@@ -0,0 +1,120 @@
+    constructor() {}
+
+    function initialize() initializer public {
+        __AccessControl_init();
+        __UUPSUpgradeable_init();
+        
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    }
+
+    /** 
+      * @notice This function sets the address of the reward contract.
+      * @dev Only the default admin can call this function.
+      * @param _reward The address of the reward contract.
+      */
+    function setRewardContract (address _reward) public onlyRole(DEFAULT_ADMIN_ROLE) {
+        rewardContract = _reward;
+    }
+
+    /** 
+      * @notice This function sets a challenge with a given struct Challenge in storage.
+      * @dev Only the default admin can call this function.
+      * @param challenge The struct Challenge that contains all necessary data for a challenge.
+      */
+    function setChallenge(Challenge calldata challenge) public onlyRole(DEFAULT_ADMIN_ROLE)  {
+        challenges[challengeIndex] = challenge;
+        challenges[challengeIndex].publishersCount = 0;
+        challenges[challengeIndex].set = 1;
+        challengeIndex++;
+    }
+
+    /** 
+      * @notice This function is used by publishers to publish their proofs for a given challenge.
+      * @param index The index of the challenge in the challenges mapping.
+      * @param proof The struct Proof containing all necessary data for a proof. This includes values a, b, c and d.
+      * @param input The array of 3 uints containing additional input data for a challenge. This includes a challengeHash, a random number (r) and a nullifier (s).
+      */
+    function publishChallenge (uint index, ZKVerifier.Proof memory proof, uint[3] memory input) public {
+        require(rewardContract != address(0), "rewardontract not set");
+        Challenge storage challenge = challenges[index];
+        require(challenge.set == 1, "challenge not set");
+        require(challenge.verifier != address(0), "no challenge started");
+        require(challenge.publishersCount < challenge.max, "publishers reached maximum amount");
+        bytes memory nullifier = abi.encodePacked(index, input[2]);
+        bytes memory publisher = abi.encodePacked(index, msg.sender);
+        require(nullifiers[nullifier] == 0, "proof already published");
+        require(publishers[publisher] == 0, "current publisher has already submitted");
+        require(challenge.challengeHash == input[1], "provided challenge is not valid");
+        
+        // function verifyTx(Proof memory proof, uint[3] memory input) public view returns (bool r)
+        (bool success, bytes memory data) = challenge.verifier.call{ value: 0 }(
+            abi.encodeWithSignature("verifyProof(uint256[2],uint256[2][2],uint256[2],uint256[3])", proof.a, proof.b, proof.c, input)
+        );
+        
+        require(success, "the call to the verifier failed");
+
+        (bool verified) = abi.decode(data, (bool));        
+        require(verified, "the provided proof isn't valid");        
```

Comment:
The code from this line to the bottom looks like it is broken, because the syntax highlight fails to highlight the code correctly. It might have to do with the message "the provided proof isn't valid" maybe use "isn\'t"